[quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51259 [quant][fx] Update name of packed weight attributes
* **#51171 [quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)**
* #51166 [quant][fx] Make scale, zero_point buffers in the model and use FQN (for quantized ops)
* #51086 [quant][fx] Scope support for call_function in QuantizationTracer

Summary:
Following up on previous PR, this PR makes scale and zero_point for quantize_per_tensor to be
registered as buffers in the module.
Currently the dtype is still stored as attr (not registered as buffer) since we can only register tensor types.

Test Plan:
python test/test_quantization.py test_qparams_buffers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26092964](https://our.internmc.facebook.com/intern/diff/D26092964)

supriyar committed:
[quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)
Summary:
Following up on previous PR, this PR makes scale and zero_point for quantize_per_tensor to be
registered as buffers in the module.
Currently the dtype is still stored as attr (not registered as buffer) since we can only register tensor types.

Test Plan:
python test/test_quantization.py test_qparams_buffers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantize_fx.py:
```diff
@@ -1527,11 +1527,13 @@ def forward(self, x):
+        quant_scale_count = quant_zero_point = scale_count = zero_point_count = 0
         for k in keys:
-            if 'scale' in k:
+            if 'quant_scale' in k:
```
nit: `quant_scale` is a bit vague, shall we rename it to `input_scale`? same for `zero_point`

jerryzh168 reviewed (approved):
thanks for cleaning up the code, please fix ci before landing

supriyar committed:
Update on "[quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)"


Summary:
Following up on previous PR, this PR makes scale and zero_point for quantize_per_tensor to be
registered as buffers in the module.
Currently the dtype is still stored as attr (not registered as buffer) since we can only register tensor types.

Test Plan:
python test/test_quantization.py test_qparams_buffers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26092964](https://our.internmc.facebook.com/intern/diff/D26092964)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)"


Summary:
Following up on previous PR, this PR makes scale and zero_point for quantize_per_tensor to be
registered as buffers in the module.
Currently the dtype is still stored as attr (not registered as buffer) since we can only register tensor types.

Test Plan:
python test/test_quantization.py test_qparams_buffers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26092964](https://our.internmc.facebook.com/intern/diff/D26092964)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)"


Summary:
Following up on previous PR, this PR makes scale and zero_point for quantize_per_tensor to be
registered as buffers in the module.
Currently the dtype is still stored as attr (not registered as buffer) since we can only register tensor types.

Test Plan:
python test/test_quantization.py test_qparams_buffers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26092964](https://our.internmc.facebook.com/intern/diff/D26092964)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)"


Summary:
Following up on previous PR, this PR makes scale and zero_point for quantize_per_tensor to be
registered as buffers in the module.
Currently the dtype is still stored as attr (not registered as buffer) since we can only register tensor types.

Test Plan:
python test/test_quantization.py test_qparams_buffers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26092964](https://our.internmc.facebook.com/intern/diff/D26092964)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)"


Summary:
Following up on previous PR, this PR makes scale and zero_point for quantize_per_tensor to be
registered as buffers in the module.
Currently the dtype is still stored as attr (not registered as buffer) since we can only register tensor types.

Test Plan:
python test/test_quantization.py test_qparams_buffers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26092964](https://our.internmc.facebook.com/intern/diff/D26092964)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)"


Summary:
Following up on previous PR, this PR makes scale and zero_point for quantize_per_tensor to be
registered as buffers in the module.
Currently the dtype is still stored as attr (not registered as buffer) since we can only register tensor types.

Test Plan:
python test/test_quantization.py test_qparams_buffers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26092964](https://our.internmc.facebook.com/intern/diff/D26092964)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)"


Summary:
Following up on previous PR, this PR makes scale and zero_point for quantize_per_tensor to be
registered as buffers in the module.
Currently the dtype is still stored as attr (not registered as buffer) since we can only register tensor types.

Test Plan:
python test/test_quantization.py test_qparams_buffers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26092964](https://our.internmc.facebook.com/intern/diff/D26092964)

[ghstack-poisoned]
